src/anemoi/datasets/create/gridded/statistics/__init__.py

```diff
@@ -0,0 +1,561 @@
+        sums[i] = np.nansum(values, axis=1)
+        squares[i] = np.nansum(values * values, axis=1)
+        count[i] = np.sum(~np.isnan(values), axis=1)
+        has_nans[i] = np.isnan(values).any()
+
+    LOG.info(f"Statistics computed for {nvars} variables.")
+
+    return {
+        "minimum": minimum,
+        "maximum": maximum,
+        "sums": sums,
+        "squares": squares,
+        "count": count,
+        "has_nans": has_nans,
+    }
+
+
+class TmpStatistics:
+    """Temporary statistics storage class."""
+
+    version = 3
+    # Used in parrallel, during data loading,
+    # to write statistics in pickled npz files.
+    # can provide statistics for a subset of dates.
+
+    def __init__(self, dirname: str, overwrite: bool = False) -> None:
+        """Initialize TmpStatistics.
+
+        Parameters
+        ----------
+        dirname : str
+            Directory name for storing statistics.
+        overwrite : bool, optional
+            Whether to overwrite existing files. Defaults to False.
+        """
+        self.dirname = dirname
+        self.overwrite = overwrite
+
+    def add_provenance(self, **kwargs: dict) -> None:
+        """Add provenance information.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional provenance information.
+        """
+        self.create(exist_ok=True)
+        path = os.path.join(self.dirname, "provenance.json")
+        if os.path.exists(path):
+            return
+        out = dict(provenance=gather_provenance_info(), **kwargs)
+        with open(path, "w") as f:
+            json.dump(out, f, default=str)
+
+    def create(self, exist_ok: bool) -> None:
+        """Create the directory for storing statistics.
+
+        Parameters
+        ----------
+        exist_ok : bool
+            Whether to ignore if the directory already exists.
+        """
+        os.makedirs(self.dirname, exist_ok=exist_ok)
+
+    def delete(self) -> None:
+        """Delete the directory for storing statistics."""
+        try:
+            shutil.rmtree(self.dirname)
+        except FileNotFoundError:
+            pass
+
+    def write(self, key: str, data: any, dates: list[datetime.datetime]) -> None:
+        """Write statistics data to a file.
+
+        Parameters
+        ----------
+        key : str
+            The key for the data.
+        data : any
+            The data to write.
+        dates : list of datetime.datetime
+            List of dates associated with the data.
+        """
+        self.create(exist_ok=True)
+        h = hashlib.sha256(str(dates).encode("utf-8")).hexdigest()
+        path = os.path.join(self.dirname, f"{h}.npz")
+
+        if not self.overwrite:
+            assert not os.path.exists(path), f"{path} already exists"
+
+        tmp_path = path + f".tmp-{os.getpid()}-on-{socket.gethostname()}"
+        with open(tmp_path, "wb") as f:
+            pickle.dump((key, dates, data), f)
```

Comment:
What is "key" here? I also see it's ignored in the aggregator `_, _dates, stats in self.owner._gather_data()`